cpp/src/search/search.cu

```diff
@@ -100,46 +100,43 @@ std::unique_ptr<column> search_ordered(table_view const& t,
 
   // This utility will ensure all corresponding dictionary columns have matching keys.
   // It will return any new dictionary columns created as well as updated table_views.
-  auto matched  = dictionary::detail::match_dictionaries({t, values}, stream);
-  auto d_t      = table_device_view::create(matched.second.front(), stream);
-  auto d_values = table_device_view::create(matched.second.back(), stream);
-  auto count_it = thrust::make_counting_iterator<size_type>(0);
-
-  rmm::device_vector<order> d_column_order(column_order.begin(), column_order.end());
-  rmm::device_vector<null_order> d_null_precedence(null_precedence.begin(), null_precedence.end());
+  auto const [_, t_vals_matched] = dictionary::detail::match_dictionaries({t, values}, stream);
+
+  auto const [t_flattened, column_order_flattened, null_precedence_flattened, __] =
+    structs::detail::flatten_nested_columns(t_vals_matched.front(), column_order, null_precedence);
+  auto const [values_flattened, ___, ____, _____] =
+    structs::detail::flatten_nested_columns(t_vals_matched.back(), {}, {});
+
+  auto const t_d      = table_device_view::create(t_flattened, stream);
+  auto const values_d = table_device_view::create(values_flattened, stream);
+  auto const& lhs     = find_first ? *t_d : *values_d;
+  auto const& rhs     = find_first ? *values_d : *t_d;
+
+  rmm::device_uvector<order> column_order_dv(column_order_flattened.size(), stream);
+  rmm::device_uvector<null_order> null_precedence_dv(null_precedence_flattened.size(), stream);
+  CUDA_TRY(cudaMemcpyAsync(column_order_dv.data(),
+                           column_order_flattened.data(),
+                           sizeof(order) * column_order_flattened.size(),
+                           cudaMemcpyDefault,
+                           stream.value()));
+  CUDA_TRY(cudaMemcpyAsync(null_precedence_dv.data(),
+                           null_precedence_flattened.data(),
+                           sizeof(null_order) * null_precedence_flattened.size(),
+                           cudaMemcpyDefault,
+                           stream.value()));
```

Comment:
Can you use the new vector factories functions for these instead? Something like:
```
auto column_order_dv = make_device_uvector_async( column_order_flattened, stream );
```
https://github.com/rapidsai/cudf/blob/05330ddad6d32d70088c7c725c41296f917359e0/cpp/include/cudf/detail/utilities/vector_factories.hpp#L125